templates/tfengine/recipes/project/data.yaml

```diff
@@ -1,35 +1,36 @@
 templates:
-- name: "dir"
+- name: "project"
+  recipe_path: "../folder/project.yaml"
+
+- name: "data_dir"
+  output_ref: "project.dir"
   output_path: "./data"
-- name: "parent_dep"
-  component_path: "../../components/terragrunt/dependency"
-  output_ref: "dir"
+
+- name: "terraform"
+  recipe_path: "../terraform/terraform.yaml"
+  output_ref: "data_dir"
   data:
-    DEPENDENCIES:
+    VARS:
+    - NAME: "project_id"
+      TYPE: "string"
+    {{if get . "STORAGE_BUCKETS"}}
+    - NAME: "storage_location"
+      TYPE: "string"
+      VALUE: "{{.STORAGE_LOCATION}}"
+    {{end}}
+    DEPS:
     - NAME: "project"
       PATH: "../project"
-- name: "var_project_id"
-  component_path: "../../components/terraform/variable"
-  output_ref: "dir"
-  data:
-    NAME: "project_id"
-    TYPE: "string"
-    VALUE: '"{{.PROJECT_ID}}"'
+      MOCK_OUTPUTS:
+        project_id: '"mock-project"'
+    INPUTS:
+      project_id: "dependency.project.outputs.project_id"
 
-{{if index . "STORAGE_BUCKETS"}}
-- name: "var_storage_location"
-  component_path: "../../components/terraform/variable"
-  output_ref: "dir"
-  data:
-    NAME: "storage_location"
-    TYPE: "string"
-    VALUE: '"{{.STORAGE_LOCATION}}"'
-{{range $i, $_ := .STORAGE_BUCKETS}}
+{{range $i, $_ := get . "STORAGE_BUCKETS"}}
 - name: "bucket-{{.NAME}}"
   component_path: "../../components/project/storage"
-  output_ref: "dir"
+  output_ref: "data_dir"
```

Comment:
Does this need to be unique? It would be the same output_ref for every bucket in the range.

Reply:
No, this tells the engine where to write all the buckets. In this case they are all in the same data deployment.